RedHat-UX/red-hat-design-tokens
# Implement Design Tokens

Closes #3 
Closes #6 

## What I did

- Added tokens in [yaml](https://learnxinyminutes.com/docs/yaml/) format
  - YAML is a superset of JSON, with a terser, more forgiving, and white-space sensitive syntax 
- Used `style-dictionary` to output multiple files and helpers:
  - global page css (`:root`)
  - component reset (`:host`)
  - scss (e.g. `$rh-color-blue-300`)
  - json (`style-dictionary` json format)
  - js (module and cjs formats, with `.d.ts` declarations)
  - vscode snippets (also usable in neovim via LuaSnips)
- legal headers (MIT) for most artifacts
- Used nunjucks and web-dev-server to show off most tokens in a page from a local dev server
- Build that same page for deploy previews
- changesets and netlify cicd / release workflow

## What I *didn't* do

- full HTML of all tokens (e.g. spacers)

## Testing Instructions
See [README.md](https://github.com/RedHat-UX/red-hat-design-tokens/tree/feat/initial-tokens#red-hat-design-tokens)
- [Deploy Preview](https://deploy-preview-2--red-hat-design-tokens.netlify.app/)
- [JSON](https://deploy-preview-2--red-hat-design-tokens.netlify.app/json/tokens.json)
- [CSS](https://deploy-preview-2--red-hat-design-tokens.netlify.app/css/global.css)
- [SCSS](https://deploy-preview-2--red-hat-design-tokens.netlify.app/scss/_variables.scss)
- [Lit CSS](https://deploy-preview-2--red-hat-design-tokens.netlify.app/css/reset.css.js)
